docs/v0.5/getting-started/kind.md

```diff
@@ -0,0 +1,307 @@
+configmaps                            allowed   allowed
+secrets                               allowed   allowed
+pods                                  allowed   n/a
+pods/log                              allowed   n/a
+applications.build.projectriff.io     allowed   allowed
+containers.build.projectriff.io       allowed   allowed
+functions.build.projectriff.io        allowed   allowed
+deployers.core.projectriff.io         allowed   allowed
+processors.streaming.projectriff.io   allowed   allowed
+streams.streaming.projectriff.io      allowed   allowed
+adapters.knative.projectriff.io       allowed   allowed
+deployers.knative.projectriff.io      allowed   allowed
+```
+
+### apply build credentials
+
+Use the riff CLI to apply credentials for a container registry. If you plan on using a namespace other than `default` add the `--namespace` flag. Replace the ??? with your docker username.
+
+```sh
+DOCKER_ID=???
+```
+
+```sh
+riff credential apply my-creds --docker-hub $DOCKER_ID --set-default-image-prefix
+```
+
+You will be prompted to provide the password.
+
+## Create a function
+
+This step will pull the source code for a function from a GitHub repo, build a container image based on the node function invoker, and push the resulting image to your Docker Hub repo.
+
+```sh
+riff function create square \
+  --git-repo https://github.com/projectriff-samples/node-square  \
+  --artifact square.js \
+  --tail
+```
+
+After the function is created, you can see the built image by listing functions.
+
+```sh
+riff function list
+```
+
+```
+NAME     LATEST IMAGE                                                                                                ARTIFACT    HANDLER   INVOKER   STATUS   AGE
+square   index.docker.io/$DOCKER_ID/square@sha256:ac089ca183368aa831597f94a2dbb462a157ccf7bbe0f3868294e15a24308f68   square.js   <empty>   <empty>   Ready    1m13s
+```
+
+## Create a Knative deployer
+
+The [Knative Runtime](../runtimes/knative.md) is only available on clusters with Istio and Knative installed. Knative deployers run riff workloads using Knative resources which provide auto-scaling (including scale-to-zero) based on HTTP request traffic, and routing.
+
+```sh
+riff knative deployer create knative-square --function-ref square --tail
+```
+
+After the deployer is created, you can see the hostname by listing deployers.
+
+```sh
+riff knative deployer list
+```
+```
+NAME             TYPE       REF      HOST                                 STATUS   AGE
+knative-square   function   square   knative-square.default.example.com   Ready    28s
+```
+
+### invoke the function
+
+Knative configures HTTP routes on the istio-ingressgateway. Requests are routed by hostname.
+
+Look up the IP address for the kind node; you should see an address like `172.17.0.3`.
+
+```sh
+KIND_NODE_IP=$(kubectl get node -o jsonpath='{$.items[0].status.addresses[?(@.type=="InternalIP")].address}')
+echo $KIND_NODE_IP
+```
+
```

Comment:
so, the ip+port doesn't seem to work on my Mac, I get `curl: (7) Failed to connect to 172.17.0.2 port 32362: Operation timed out`

Reply:
docker containers in general do not have reachable IPs on the host with docker for mac :(
you have to use a port forward. kind can add these to a node via config, e.g. https://kind.sigs.k8s.io/docs/user/resources/#using-contour-ingress-with-kind

Reply:
Got this working now using extraPortMappings as per the latest kind Ingress docs
https://kind.sigs.k8s.io/docs/user/ingress/